### ISSUE TITLE
Rewrite rendering stuff

### ISSUE THREAD

**Issue by myamusashi** [KZDKM/Hyprspace#131]
https://github.com/hyprwm/Hyprland/commit/e536b02248f370971716b744968776f6880528be
Just marking here, to rewrite the rendering schedule. I'd like to help, but either the new rendering schedule is hard to understand (I've read some implementations in other plugins, it's pretty easy, maybe) or the codebase of this plugin is hard to understand, maybe the way this plugin renders things is different from others.

**Comment by myamusashi** [comment]
![2024-12-30_21-12-1735567942](https://github.com/user-attachments/assets/f76a876a-5336-4fee-a2ab-d69ede593eba)
Yup, skill issue here

**Comment by alba4k** [comment]
windows just disappear on my end when trying to open the overview lol

**Comment by myamusashi** [comment]
https://github.com/hyprwm/Hyprland/issues/8918 Same issue?

**Comment by myamusashi** [comment]
![image](https://github.com/user-attachments/assets/c47cbe0a-bb96-408e-b2d1-315f9e1bd125)
Oh nice, this is a bug apparently, but a lot of damage because I haven't changed the rendering for rect and border.

**Comment by myamusashi** [comment]
> windows just disappear on my end when trying to open the overview lol

You can use the plugin on the latest -git

**Comment by myamusashi** [comment]
https://github.com/KZDKM/Hyprspace/pull/133 You can test it

**Comment by KZDKM** [comment]
Bruh.

The endless refactoring upstream istg... Sorry for being inactive cuz im on vacation rn. I will look at it when I have access to my main machine again.

**Comment by myamusashi** [comment]
All good

**Comment by 0xl30** [comment]
Hello Bro,
Have you been able to fix the 0.47.0 issue..?
And how can I install your fix..?

**Comment by myamusashi** [comment]
> Have you been able to fix the 0.47.0 issue..?

Nah

![Image](https://github.com/user-attachments/assets/3e6699a4-0608-4736-b549-ecc79f3cae19)

**Comment by Beast-Viper** [comment]
Commenting to keep activity.

**Comment by PokumeKachi** [comment]
Commenting to keep activity.

**Comment by KZDKM** [comment]
Sorry for the delay. Started working on this yesterday as I updated to 0.47. Currently the plugin would build and run on my end. But the windows on my secondary monitor cuts off for some reason. Please can yall test it and see what works and what doesn't.

**Comment by myamusashi** [comment]
I can test it

**Comment by KZDKM** [comment]
So uhh, the windows would cut off for any monitor with scale=/=1. Weird.

**Comment by myamusashi** [comment]
![Image](https://github.com/user-attachments/assets/f0308851-8b30-4514-8fa1-ad9a5cdfa1e9)

https://github.com/user-attachments/assets/c51e6055-72ca-493f-88ce-017ade36be79

For zen-browser, the window is always cut but for mpv, pavucontrol, etc. there is a bug that if not in focus (the window) will be visible.

**Comment by KZDKM** [comment]
Can you set debug::pass=true in hyprland config and take a screenshot?

**Comment by myamusashi** [comment]
![Image](https://github.com/user-attachments/assets/1f40a649-8459-4884-99e5-86130ab27d40)
![Image](https://github.com/user-attachments/assets/a24e2be6-3186-46bc-b317-b603565b76c9)
![Image](https://github.com/user-attachments/assets/ff8fc8eb-d47b-4ece-a547-10582e253af0)

**Comment by KZDKM** [comment]
On my end it works perfectly for any unscaled monitor. Windows cutting off is a side effect of calling renderwindow twice but i have no idea why it is causing that problem. I doubt its even related to the new occlusion optimization.

**Comment by KZDKM** [comment]
@myamusashi  could you try again with the newest commit?

**Comment by myamusashi** [comment]
Nope, not working. I'm trying to update my system for now maybe the result will be different..maybe

**Comment by 0xl30** [comment]
> Nope, not working. I'm trying to update my system for now maybe the result will be different..maybe

hello bro can you tell me how showing waybar when you using hyprspace.?

**Comment by myamusashi** [comment]
> > Nope, not working. I'm trying to update my system for now maybe the result will be different..maybe
> 
> hello bro can you tell me how showing waybar when you using hyprspace.?

Change the layer to top or overlay

**Comment by KZDKM** [comment]
The fix was to create a fake pass that overlays a damage region on the monitor. This essentially prevents anything on top of the screen to be optimized out whenever the overview is open. After the fix its working just as before now so I cant repro the issue anymore. Is the behavior exactly the same as before?

**Comment by myamusashi** [comment]
After updating my system, this plugin works perfectly now.

**Comment by myamusashi** [comment]
Should we close this?

**Comment by KZDKM** [comment]
Ok great. I should now nudge the hyprlandplugins nixpkgs maintainers to unflag this plugin as broken.

**Comment by 0xl30** [comment]
> Change the layer to top or overlay

please tell me in details

> overlay

you mean "plugin:overview:onTop" ?? please tell me in details.
